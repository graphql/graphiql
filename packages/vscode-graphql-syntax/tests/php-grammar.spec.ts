import { tokenizeFile } from './__utilities__/utilities';

describe('inline.graphql.php grammar', () => {
  const scope = 'inline.graphql.php';

  it('should tokenize a simple php file', async () => {
    const result = await tokenizeFile('__fixtures__/test.php', scope);
    expect(result).toMatchSnapshot();
  });
});
