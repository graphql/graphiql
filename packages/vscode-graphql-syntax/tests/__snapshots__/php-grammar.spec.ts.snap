// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`inline.graphql.php grammar > should tokenize a simple php file 1`] = `
<?php                          | 
                               | 
## TODO: not yet supported     | 
$graphql = graphql([           | 
    'query' => "{              | 
  post(idType: SLUG, id: 2) {  | 
    title                      | 
    content                    | 
    date                       | 
  }                            | 
}"                             | 
]);                            | 
                               | 
$query =                       | 
<<<                            | punctuation.section.embedded.begin.php punctuation.definition.string.php
GRAPHQL                        | punctuation.section.embedded.begin.php keyword.operator.heredoc.php
                               | meta.embedded.block.graphql
query                          | meta.embedded.block.graphql keyword.operation.graphql
                               | meta.embedded.block.graphql meta.selectionset.graphql
{                              | meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
                               | meta.embedded.block.graphql meta.selectionset.graphql
site                           | meta.embedded.block.graphql meta.selectionset.graphql variable.graphql
                               | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
{                              | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
                               | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
name                           | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql variable.graphql
                               | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
}                              | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
                               | meta.embedded.block.graphql meta.selectionset.graphql
}                              | meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
GRAPHQL                        | punctuation.section.embedded.end.php keyword.operator.heredoc.php
;                              | 
                               | 
## TODO: not yet supported     | 
$gql = <<<QUERY                | 
query {                        | 
    pokemon(name: "Pikachu") { | 
        id                     | 
        number                 | 
        name                   | 
        attacks {              | 
            special {          | 
                name           | 
                type           | 
                damage         | 
            }                  | 
        }                      | 
    }                          | 
}                              | 
QUERY;                         | 
                               | 
?>                             | 
                               | 
`;
